packages/ethereum/contracts/HoprChannels.sol

```diff
@@ -183,12 +189,15 @@ contract HoprChannels is IERC777Recipient, ERC1820Implementer, Multicall {
     }
 
     /**
-     * @dev Announces msg.sender's multiaddress.
-     * Confirmation should be done off-chain.
+     * @dev Announces msg.sender's publicKey and multiaddress.
+     * Multiaddress confirmation should be done off-chain.
+     * @param publicKey the msg.sender's public key
      * @param multiaddr the multiaddress
      */
-    function announce(bytes calldata multiaddr) external {
-        emit Announcement(msg.sender, multiaddr);
+    function announce(bytes calldata publicKey, bytes calldata multiaddr) external {
+        require(address(uint160(uint256(keccak256(publicKey)))) == msg.sender, "publicKey's address does not match senders");
```

Comment:
This does not check the multiaddr, hence pubkey and the pubkey in the multiaddr can be different and can create inconsistencies

Reply:
how would you verify the pubkey in the multiaddr?

Reply:
I *would not* do it because I consider this way too complicated in the smart contract. Here are the reasons:
- hopr addresses are given as *compressed* pubkeys and decompressing pubkeys is infeasible in SCs.
- if someone *intentionally* announces wrong information, that's their choice. But we can enforce *that* people announce

Other solution:
change announce to `announce(bytes ipPort, bytes pubKey)` and emit `Announcement(ipPort, pubkey)` and off-chain you can rebuild the Multiaddr (which can be done easily using Multiaddr package) from ipPort and pubKey
In this case, we can *check* the public key on-chain but we *need* to pass it uncompressed. Nevertheless this option is way too much engineering for the problem.